ee/localserver/krypto-ec-middleware.go

```diff
@@ -216,6 +218,41 @@ func (e *kryptoEcMiddleware) Wrap(next http.Handler) http.Handler {
 			}()
 		}
 
+		// Check if the origin is in the allowed list. See https://github.com/kolide/k2/issues/9634
+		if len(cmdReq.AllowedOrigins) > 0 {
+			allowed := false
+			for _, ao := range cmdReq.AllowedOrigins {
+				if strings.EqualFold(ao, r.Header.Get("Origin")) {
+					allowed = true
+					break
+				}
+			}
+
+			if !allowed {
+				span.SetAttributes(attribute.String("origin", r.Header.Get("Origin")))
+				span.SetStatus(codes.Error, "origin is not allowed")
```

Comment:
```suggestion
				traces.SetError(span, fmt.Errorf("origin %s is not allowed", r.Header.Get("Origin")))
```

`SetError` will set the status on the span plus make some other attempts to capture info

Reply:
Oh right! We have a helper. I updated the other parts of the file too